From fe9286ea52d68e3f7295938ef37a44b82719f880 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:25:02 -0400
Subject: [s390] pci: Remove superfluous check of MSI type

Message-id: <20140719042502.19438.5147.stgit@gir.stowe>
Patchwork-id: 85984
O-Subject: [RHEL7.1 PATCH 22/81] s390/PCI: Remove superfluous check of MSI type
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: 57b7cb024422bbe85366ef28f5e192997bd16943

commit 57b7cb024422bbe85366ef28f5e192997bd16943
Author: Alexander Gordeev <agordeev@redhat.com>
Date:   Mon Dec 16 09:34:55 2013 +0100

    s390/PCI: Remove superfluous check of MSI type

    arch_setup_msi_irqs() hook can only be called from the generic MSI code
    which ensures correct MSI type parameter.

    Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Acked-by: Martin Schwidefsky <schwidefsky@de.ibm.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/s390/pci/pci.c b/arch/s390/pci/pci.c
index 189b49c..7b4f2ea 100644
--- a/arch/s390/pci/pci.c
+++ b/arch/s390/pci/pci.c
@@ -436,8 +436,6 @@ int arch_setup_msi_irqs(struct pci_dev *pdev, int nvec, int type)
  struct msi_desc *msi;
  int rc;
 
- if (type != PCI_CAP_ID_MSIX && type != PCI_CAP_ID_MSI)
-  return -EINVAL;
  if (type == PCI_CAP_ID_MSI && nvec > 1)
   return 1;
  msi_vecs = min(nvec, ZPCI_MSI_VEC_MAX);
-- 
1.7.1